internal/planner/planner_test.go

```diff
@@ -224,9 +225,41 @@ func TestPlannerHelloWorld(t *testing.T) {
 			note:    "relation unify",
 			queries: []string{`walk(input, [["foo", y], x])`},
 		},
+		{
+			note:    "cross product with non-ground refs to packages (simplified)",
+			queries: []string{`x := "q"; y := data.foo[x]`},
+			modules: []string{`
+		package foo
+p = 1
+q = 2`},
+		},
+		{
+			note:    "with undo data shadow",
+			queries: []string{`data.test.p = x; data.test.q = y with data.test.r as 2; data.test.r = z; data.test.q = t`},
+			modules: []string{`
+		package test
+		p = data.test.r
+		q = [data.test.r]`},
+		},
+		{
+			note:    "with virtual doc iteration",
+			queries: []string{`x := data[i][j] with data.bar.p as 3 with data.bar.q as 4; y = data.bar.p; z = data.bar.q`},
+			modules: []string{`
+		package foo
+p = 0
+q = 1
+r = 2`,
+`package bar
+p = -1
+`,
```

Comment:
☝️ This makes me thing that I might be missing some recursive call in the planner, we only get one function generated...